public/app/features/explore/Logs/Logs.tsx

```diff
@@ -334,18 +342,51 @@ const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
       panelState?.logs?.columns,
       panelState?.logs?.displayedFields,
       panelState?.logs?.refId,
+      panelState?.logs?.tableSortBy,
+      panelState?.logs?.tableSortDir,
       visualisationType,
     ]
   );
 
+  // Sync between local displayedFields and panelState and show original line button
   useEffect(() => {
-    if (!shallowCompare(displayedFields, panelState?.logs?.displayedFields ?? [])) {
+    const panelDisplayedFields = panelState?.logs?.displayedFields ?? [];
+
+    // In table mode always sync with panelState
+    if (visualisationType === 'table') {
+      setDisplayedFields(panelDisplayedFields);
+      return;
+    }
+
+    // Only sync when transitioning from 'table' to 'logs'
+    if (previousVisualisationType !== 'table' || visualisationType !== 'logs') {
```

Comment:
While I'm not seeing the runtime error anymore, I'm not seeing the sync between table and logs working any more

https://github.com/user-attachments/assets/5a9ee2b0-bb50-486e-a836-6c68f67fc860

Also now that the logs panel in main has a dedicated field for the log line can we sync "Log line" in the logs panel with "Body" in the table?

Also the icons are not rendering in the table any more:

<img width="1989" height="674" alt="Image" src="https://github.com/user-attachments/assets/546af1a7-afd4-48e9-ab39-a033194ba1e9" />
I checked main and they're still working there

Reply:
TY! Okay I fixed the re-rendering error. I tried to sync the line and time, but ended up creating more 🐛 . I'm going to continue on with Table this quarter [ticket](https://github.com/grafana/logs-drilldown/issues/1462) and I'm tracking new requirements. Once this PR gets closed out I'll make new tickets.